superset/views.py

```diff
@@ -2199,10 +2199,17 @@ def sqllab_viz(self):
                 dims.append(col)
             agg = config.get('agg')
             if agg:
-                metrics.append(models.SqlMetric(
-                    metric_name="{agg}__{column_name}".format(**locals()),
-                    expression="{agg}({column_name})".format(**locals()),
-                ))
+                if agg == 'count_distinct':
```

Comment:
sure, just make sure that this call:
https://github.com/airbnb/superset/blob/7eef46e9413b01ec15be515567a9619c695d5501/superset/models.py#L1255
references your implementation